client/src/deploy/creation_common.rs

```diff
@@ -536,7 +537,11 @@ pub(super) fn apply_common_creation_options<'a, 'b>(
         .arg(ttl::arg())
         .arg(gas_price::arg())
         .arg(dependencies::arg())
-        .arg(chain_name::arg());
+        .arg(chain_name::arg())
+        .arg(
+            common::session_account::arg(DisplayOrder::SessionAccount as usize)
+                .required_unless(common::secret_key::ARG_NAME),
```

Comment:
I don't think this is correct.  We already specified that this arg is not required, and I think it indeed shouldn't be required, regardless of whether the secret key arg is provided or not.

Reply:
Changed